tt_metal/impl/device/device.hpp

```diff
@@ -118,10 +118,20 @@ class Device {
         return tt::Cluster::instance().get_active_ethernet_cores(this->id_, skip_reserved_tunnel_cores);
     }
 
+    bool is_active_ethernet_core(CoreCoord logical_core, bool skip_reserved_tunnel_cores=false) const {
+        return tt::Cluster::instance().get_active_ethernet_cores(this->id_, skip_reserved_tunnel_cores).find(logical_core)
```

Comment:
it's not clear to me that the compiler will optimize away the 2nd lookup, grab the value and use it twice:
tmp = tt::Cluster::instance().get_active_ethernet_cores(this->id_, skip_reserved_tunnel_cores);
tmp.find() != tmp.last();

Reply:
Fixed